src/main/scala/com/horizon/exchangeapi/PatternsRoutes.scala

```diff
@@ -659,20 +661,30 @@ trait PatternRoutes extends ScalatraBase with FutureSupport with SwaggerSupport
         archList += ""
         archList += "*"
         val archSet = archList.toSet
-        // archList.contains(n.arch.toString()
         if (found) {
-          if (svcArch == "" || svcArch == "*" || archSet("") || archSet("*")){
+          // if (the arch was not passed in AND one of the arches found was wildcard) OR the arch passed in was * OR the arch passed in was ""
+          if (((svcArch == "" || svcArch == "*" || archSet("") || archSet("*")) && selectedServiceArch.isEmpty) || selectedServiceArch.equals(Some("*")) || selectedServiceArch.equals(Some(""))){
```

Comment:
It is probably because you were modifying existing code, but the logic in this whole section seems more complicated than it needs to be. It is often useful to first write a block comment that clearly/succinctly describes the logic to others reading the code. That is not only useful to them, but also helps you write the code more clearly. I think the logic precedence is:

1. if the caller specified a non-wildcard arch in the body, that trumps everything, so filter on that arch
1. else if the caller or any service specified a blank/wildcard arch, then don't filter on arch at all
1. else filter on the arches in the services (like before)